primo-explore/custom/01MIT_INST-TACOS/js/custom.js

```diff
@@ -14,8 +42,56 @@
       '<div id="title-mit"><a href="https://mit.primo.exlibrisgroup.com/discovery/search?vid=01MIT_INST:MIT&lang=en">Search Our Collections</a></div>',
   });
   app.component("prmSearchBarAfter", {
-    template: alertBannerTemplate,
+    template: `<tacos></tacos>${alertBannerTemplate}`,
+  });
+  app.component("tacos", {
+    controller: function tacosController($scope, tacosService, $stateParams) {
+      // $stateParams seems to behave like an angularJS service, but I'm not sure
+      // where in the Primo VE source code it is defined...
+      // We are able to inject it in the controller and then get access to the .query property
+      // which is the raw search string(s) we want to send to TACOS
+
+      var vm = this;
+
+      function parseSearchQuery(searchQuery) {
+        // Returns searchQuery if it is a string.
+        // If searchQuery is an array of strings, joins the strings and returns the result
+
+        if (typeof searchQuery === "string") { // handle a string (simple search)
+          return searchQuery;
+        }
+        return searchQuery.join(" "); // handle an array of strings (advanced search)
+      }
+
+      $scope.$watch(function () {
+        // Watch for changes to the $stateParams.query property
+
+        return parseSearchQuery(
+          $stateParams.query,
+        );
+      }, function (newSearchQuery) {
+        // This listener function triggers when the watcher is first initialized and
+        // each time the results of the watched expression change afterward.
+        // During the initial trigger, the listener will send a search query to TACOS
+        // if the client's initial request to Primo includes search parameters.
+        // This behavior is desirable because it ensures that TACOS receives a search query
+        // right from the start. Without it, TACOS would only receive queries on subsequent user searches.
+
+        if (newSearchQuery) { //avoid sending 'undefined' searches to TACOS
+          tacosService.sendQueryToTacos(newSearchQuery).then(
+            function (response) {
+              console.log(response);
+              vm.searchTerms = response.data.data.logSearchEvent.phrase;

```

Comment:
Naming things is hard, and this may be a nit that doesn't need picked, but I wonder whether `searchTerms` is the right variable name here. At various points this PR uses both `searchTerm` and `searchTerms` - one seems to be information taken from Primo and being sent to TACOS, while another seems to be used for information returning from TACOS.

If I understand the structure of this component correctly, could we update `searchTerms` here and on line 92 below (which I think are all the same thing) to be more clearly a response from TACOS - maybe `tacosPhrase` or something?